Source/DafnyCore/Backends/Library/LibraryBackend.cs

```diff
@@ -72,13 +72,6 @@ public class LibraryBackend : ExecutableBackend {
       throw new UnsupportedFeatureException(dafnyProgram.GetStartOfFirstFileToken(), Feature.LegacyCLI);
     }
 
-    var disallowedAssumptions = dafnyProgram.Assumptions(null)
-      .Where(a => !a.desc.allowedInLibraries);
-    foreach (var assumption in disallowedAssumptions) {
-      var message = assumption.desc.issue.Replace("{", "{{").Replace("}", "}}");
-      Reporter.Error(MessageSource.Compiler, assumption.tok, message, message);
-    }
-
```

Comment:
I might be missing something, but since the LibraryBackend doesn't use the SinglePassCodeGenerator, how does it emit any warnings when building? I made use of `Assumptions()` because it was a much simpler way to check the program for things that should be rejected when building a library.

Reply:
I believe all relevant warnings and errors are emitted by the resolver, not by `SinglePassCodeGenerator`, so it doesn't matter that the library back-end doesn't use `SinglePassCodeGenerator`. The test you mention, `separate-verification/assumptions.dfy`, seems to give the right results for this PR.

Reply:
Ah great! I think my understanding of where warnings/errors were emitted was out of date - at least at one point things like bodiless declarations were only caught by the `SinglePassCodeGenerator`. Nice work correcting that in earlier changes so we can make this change!

Reply:
Thanks